### PR TITLE
[Dataset] Exclude breaking test case in `read_parquet_benchmark_single_node` release test

### LINKED ISSUE #31888: [Ray 2.3 release] [data] `read_parquet_benchmark_single_node` release test failure
https://buildkite.com/ray-project/release-tests-branch/builds/1303#0185d157-4e6a-4e4e-9629-8badcc674ecc

```
Traceback (most recent call last):
  File "read_parquet_benchmark.py", line 88, in <module>
    run_read_parquet_benchmark(benchmark)
  File "read_parquet_benchmark.py", line 54, in run_read_parquet_benchmark
    filter=filter_expr,
  File "/home/ray/oss-release-tests-2/benchmark.py", line 49, in run
    output_ds = fn(**fn_run_args)
  File "read_parquet_benchmark.py", line 25, in read_parquet
    columns=columns,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/read_api.py", line 499, in read_parquet
    **arrow_parquet_args,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/data/read_api.py", line 311, in read_datasource
    _wrap_arrow_serialization_workaround(read_args),
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/remote_function.py", line 129, in _remote_proxy
    return self._remote(args=args, kwargs=kwargs, **self._default_options)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/tracing/tracing_helper.py", line 307, in _invocation_remote_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/remote_function.py", line 412, in _remote
    return invocation(args, kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/remote_function.py", line 399, in invocation
    serialized_runtime_env_info or "{}",
  File "python/ray/_raylet.pyx", line 1968, in ray._raylet.CoreWorker.submit_task
  File "python/ray/_raylet.pyx", line 1972, in ray._raylet.CoreWorker.submit_task
  File "python/ray/_raylet.pyx", line 424, in ray._raylet.prepare_args_and_increment_put_refs
  File "python/ray/_raylet.pyx", line 415, in ray._raylet.prepare_args_and_increment_put_refs
  File "python/ray/_raylet.pyx", line 462, in ray._raylet.prepare_args_internal
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/serialization.py", line 450, in serialize
    return self._serialize_to_msgpack(value)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/serialization.py", line 429, in _serialize_to_msgpack
    metadata, python_objects
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/serialization.py", line 390, in _serialize_to_pickle5
    raise e
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/serialization.py", line 386, in _serialize_to_pickle5
    value, protocol=5, buffer_callback=writer.buffer_callback
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 88, in dumps
    cp.dump(obj)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 733, in dump
    return Pickler.dump(self, obj)
  File "pyarrow/io.pxi", line 1090, in pyarrow.lib.Buffer.__reduce_ex__
AttributeError: module 'pickle' has no attribute 'PickleBuffer'
```

### PR DESCRIPTION
**Pull request by scottjlee**
Signed-off-by: Scott Lee <sjl@anyscale.com>

## Why are these changes needed?

The release test `read_parquet_benchmark_single_node` [fails](https://buildkite.com/ray-project/release-tests-branch/builds/1303#0185d157-4e6a-4e4e-9629-8badcc674ecc), due to using Python 3.7 and not having the `pickle5` package installed. A similar issue is discussed in https://github.com/ray-project/ray/issues/26225. We found that the test failure is contained to the portion which tests a Dataset with a filter expression (the error is related to pickling with this filter expression).

Therefore, we will temporarily disable this portion of the test, while keeping the rest of the release test (which I verified passes on the same cluster). We can come back to this in the future and fix the case with filter. Example of release test [successfully running](https://buildkite.com/ray-project/release-tests-pr/builds/26615#0185f12c-afd5-4fb4-9d47-7d9d54c8bfcd) with the filter case removed.

## Related issue number

Closes #31888 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by amogkam** [comment]
should this be added as a `ray[data]` dependency for python<=3.7?

**Comment by cadedaniel** [comment]
Can we get a review of this? This is a release blocker. cc @clarkzinzow @c21 @amogkam @jianoaix

**Comment by cadedaniel** [comment]
@scottjlee can you try running this manually (i.e. fire off a release test before merging) to confirm that it fixes the issue?

**Comment by amogkam** [comment]
If this is a problem that users will run into, then we should update setup.py and add pickle5 as a dependency for the ray[data] install

**Comment by jianoaix** [comment]
Yeah, looks like the test actually exposed a prod issue for users that we need to fix.

**Comment by scottjlee** [comment]
I'm waiting on getting access to AWS roles/credentials to run release tests locally. Is there another way to kickoff release tests without merging? @cadedaniel 

In the meantime, @amogkam -- are the changes I made in `setup.py` what you had in mind?

**Comment by scottjlee** [comment]
Looks like the failing release test passes now: https://buildkite.com/ray-project/release-tests-pr/builds/26524

**Review by c21 — APPROVED** [review]

**Review by jianoaix — APPROVED** [review]

**Review by clarkzinzow — APPROVED** [review]

File: python/setup.py
Comment (clarkzinzow) [inline python/setup.py:229]: Ah this should be `<=` right?
```suggestion
    if sys.version_info <= (3, 7):
```

**Comment by scottjlee** [comment]
We ran the failing release test on a [cluster with the same environment](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_hwrrm4esjsw1wt1wm7paamisys?command-history-section=head_start_up_log) (notably Python 3.7), and determined that the failure is contained to the portion which tests a Dataset with a filter expression (the error is related to pickling with this filter expression). 

Therefore, we will temporarily disable this portion of the test, while keeping the rest of the release test (which I verified passes on the same cluster). We can come back to this in the future and fix the case with filter, but should not be a release blocker now.

**Review by jianoaix — APPROVED** [review]
Thank you @scottjlee!

**Comment by scottjlee** [comment]
Once the release test run passes, this should be good to merge!
https://buildkite.com/ray-project/release-tests-pr/builds/26615#0185f0b2-2049-43b1-b6da-d8d4015c7327

**Comment by amogkam** [comment]
Thanks @scottjlee! can we update the title and the pr description as well

File: release/nightly_tests/dataset/read_parquet_benchmark.py
Comment (scottjlee) [reply release/nightly_tests/dataset/read_parquet_benchmark.py:40]: Added more details about Python version and pickle5 in the comments.

**Pull request merged**
